src/GraphQL.Analyzers/FieldBuilderAnalyzer.cs

```diff
@@ -0,0 +1,121 @@
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace GraphQL.Analyzers;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public class FieldBuilderAnalyzer : DiagnosticAnalyzer
+{
+    // Violation: Field<T>("name", "description", ...)
+    // Fixed:     Field<T>("name").Description("description")...
+    public static readonly DiagnosticDescriptor DoNotUseObsoleteFieldMethods = new(
+        id: "GQL004",
+        title: "Don't use obsolete Field methods",
+        messageFormat: "Use correct Field method overload",
```

Comment:
This is the message that the user sees in the build output and warning window. Note it can include placeholders that need to be filled when reporting diagnostics.